File: Exercism/java/beer-song/src/main/java/BeerSong.java
public class BeerSong {
    public String verse(int x) {
        switch (x) {
            case 0:
                return "No more bottles of beer on the wall, no more bottles of beer.\n" +
                        "Go to the store and buy some more, 99 bottles of beer on the wall.\n\n";

            case 1:
                return "1 bottle of beer on the wall, 1 bottle of beer.\n" +
                        "Take it down and pass it around, no more bottles of beer on the wall.\n\n";

            default:
                return x + " bottles of beer on the wall, " + x + " bottles of beer.\n" +
                        "Take one down and pass it around, " + (x-1) + " bottle" +
                        ((x-1 == 1 ) ? "" : "s") + " of beer on the wall.\n\n";
        }

    }

    public String sing(int from, int to) {
        StringBuilder result = new StringBuilder();

        for (int i = from ; i >= to ; --i) {
            result.append(verse(i));
        }

        return result.toString();
    }

    public String singSong() {
        return sing(99, 0);
    }
}

File: Exercism/c/beer-song/src/beer_song.c
#include <string.h>
#include <stdio.h>
#include "beer_song.h"

void verse(char* buffer, int x) {
    switch (x) {
    case 0:
        strcpy(buffer,
               "No more bottles of beer on the wall, no more bottles of beer.\n"
               "Go to the store and buy some more, "
               "99 bottles of beer on the wall.\n");
        break;

    case 1:
        strcpy(buffer,
               "1 bottle of beer on the wall, 1 bottle of beer.\n"
               "Take it down and pass it around, "
               "no more bottles of beer on the wall.\n");
        break;

    case 2:
        strcpy(buffer, "2 bottles of beer on the wall, 2 bottles of beer.\n"
                "Take one down and pass it around, "
                "1 bottle of beer on the wall.\n");
        break;

    default:
        sprintf(buffer, "%d bottles of beer on the wall, %d bottles of beer.\n"
                "Take one down and pass it around, "
                "%d bottles of beer on the wall.\n",
                x, x, x-1);
        break;
    }
}

void sing(char* buff, int from, int to) {
    strcpy(buff, "");

    char temp[1024];
    for (int i = from ; i >= to ; --i) {
        verse(temp, i);
        strcat(buff, temp);
        if (i != to)
            strcat(buff, "\n");
    }
}
